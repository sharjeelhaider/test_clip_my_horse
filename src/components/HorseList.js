import React from 'react';
import PropTypes from 'prop-types';
import {FlatList} from 'react-native';

import HorseItem from './HorseItem';

const HorseList = props => {
  const {horses} = props;
  return (
    <FlatList
      data={horses}
      renderItem={({item}) => <HorseItem horse={item} />}
      keyExtractor={(item, index) => index.toString()}
      windowSize={10}
      initialNumToRender={6}
      maxToRenderPerBatch={3}
    />
  );
};

HorseList.propTypes = {
  horses: PropTypes.array.isRequired,
};

export default HorseList;
