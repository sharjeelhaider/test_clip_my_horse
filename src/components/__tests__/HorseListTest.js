import React from 'react';
import HorseList from '../HorseList';
import mockedHorses from '../../../config/jest/dataMock';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders a HorseList using Snapshots', () => {
  const wrapper = renderer.create(<HorseList horses={mockedHorses} />);
  expect(wrapper).toMatchSnapshot();
});
