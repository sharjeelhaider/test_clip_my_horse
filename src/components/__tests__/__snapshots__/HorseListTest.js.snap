// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a HorseList using Snapshots 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": "df579a48-a7ca-4afc-89d0-caabcfac87d7",
        "name": "American Beauty",
        "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/51e87826c943fd896385734287bbac27fd43c74a.jpeg",
      },
      Object {
        "id": "8b1b98fe-077f-4faf-bb16-f088f3b6dfdf",
        "name": "Arabian Night",
        "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/dae8d44c97b286e627a62569c2ff46bbb4575270.jpeg",
      },
      Object {
        "id": "49092d28-2d7b-4774-8a40-7c1a9d3e7634",
        "name": "Autumn Gold",
        "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/b05678d2169739f280993fb8636ba017ff9b6787.jpeg",
      },
      Object {
        "id": "be11640d-1fe6-4d73-a664-0fd84ea6338c",
        "name": "Ancient Orange",
        "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/5ea73acc5924c18bec341e3b30eb6ba8ea463d28.jpeg",
      },
      Object {
        "id": "a04e73cf-6ceb-40e9-969e-e94b02f4941e",
        "name": "Amazing Grace",
        "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/7d860756e5c7fc9df00b39b459b868a12edbac9a.jpeg",
      },
      Object {
        "id": "a3c4c44c-1975-4d6d-a870-af9cd5d23628",
        "name": "Apple Candy",
        "picture": "https://d2vtwgzbrv2xnd.cloudfront.net/a7acc104b765cf17c4adeabaa2ca401ace08a5df.jpeg",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={6}
  keyExtractor={[Function]}
  maxToRenderPerBatch={3}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={10}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 2,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://d2vtwgzbrv2xnd.cloudfront.net/51e87826c943fd896385734287bbac27fd43c74a.jpeg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 150,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "paddingBottom": 4,
              }
            }
          >
            American Beauty
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icon_next.png",
              }
            }
            style={
              Object {
                "alignSelf": "flex-end",
                "height": 40,
                "width": 40,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "height": 2,
                "marginTop": 2,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 2,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://d2vtwgzbrv2xnd.cloudfront.net/dae8d44c97b286e627a62569c2ff46bbb4575270.jpeg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 150,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "paddingBottom": 4,
              }
            }
          >
            Arabian Night
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icon_next.png",
              }
            }
            style={
              Object {
                "alignSelf": "flex-end",
                "height": 40,
                "width": 40,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "height": 2,
                "marginTop": 2,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 2,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://d2vtwgzbrv2xnd.cloudfront.net/b05678d2169739f280993fb8636ba017ff9b6787.jpeg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 150,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "paddingBottom": 4,
              }
            }
          >
            Autumn Gold
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icon_next.png",
              }
            }
            style={
              Object {
                "alignSelf": "flex-end",
                "height": 40,
                "width": 40,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "height": 2,
                "marginTop": 2,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 2,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://d2vtwgzbrv2xnd.cloudfront.net/5ea73acc5924c18bec341e3b30eb6ba8ea463d28.jpeg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 150,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "paddingBottom": 4,
              }
            }
          >
            Ancient Orange
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icon_next.png",
              }
            }
            style={
              Object {
                "alignSelf": "flex-end",
                "height": 40,
                "width": 40,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "height": 2,
                "marginTop": 2,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 2,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://d2vtwgzbrv2xnd.cloudfront.net/7d860756e5c7fc9df00b39b459b868a12edbac9a.jpeg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 150,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "paddingBottom": 4,
              }
            }
          >
            Amazing Grace
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icon_next.png",
              }
            }
            style={
              Object {
                "alignSelf": "flex-end",
                "height": 40,
                "width": 40,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "height": 2,
                "marginTop": 2,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 2,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://d2vtwgzbrv2xnd.cloudfront.net/a7acc104b765cf17c4adeabaa2ca401ace08a5df.jpeg",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 150,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginLeft": 10,
                "paddingBottom": 4,
              }
            }
          >
            Apple Candy
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icon_next.png",
              }
            }
            style={
              Object {
                "alignSelf": "flex-end",
                "height": 40,
                "width": 40,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "height": 2,
                "marginTop": 2,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
